src/main/java/upeldev/com/github/upel3/services/ActivityService.java

```diff
@@ -99,4 +104,16 @@ public void removeSubActivity(Activity activity, SubActivity subActivity){
         activityRepository.save(activity);
     }
 
+    public void addSubActivity(Activity activity, SubActivity subActivity){
+        subActivityRepository.save(subActivity);
+        activity.getSubActivities().add(subActivity);
+        activity.getGrades().forEach(
+                grade -> {
+                    SubGrade subGrade = new SubGrade(subActivity, grade, 0.0);
+                    grade.getSubGrades().add(subGrade);
```

Comment:
Shouldn't subgrades be saved to repository?

subGradeService.save(subGrade);

Reply:
As far as I am concerned, it works fine. Take a look at its usage in my commit below